engine/src/main/java/org/entando/entando/web/page/PageController.java

```diff
@@ -189,15 +187,42 @@ public void setAuthorizationService(PageAuthorizationService authorizationServic
             throw new ValidationGenericException(bindingResult);
         }
 
+        PagePositionRequest pagePositionRequest = new PagePositionRequest();
+        pagePositionRequest.setParentCode(pageRequest.getParentCode());
+        pagePositionRequest.setCode(pageCode);
+        int position = pageService.getPages(pageCode).size() + 1;
+        pagePositionRequest.setPosition(position);
+
+        validateMovePage(pageCode, bindingResult, pagePositionRequest);
+
         PageDto page = this.getPageService().updatePage(pageCode, pageRequest);
         Map<String, String> metadata = new HashMap<>();
         return new ResponseEntity<>(new RestResponse<>(page, metadata), HttpStatus.OK);
     }
 
+    private void validateMovePage(@PathVariable String pageCode,
+            BindingResult bindingResult, PagePositionRequest pagePositionRequest) {
+        this.getPageValidator().validateChangePositionRequest(pageCode, pagePositionRequest, bindingResult);
+
+        if (bindingResult.hasErrors()) {
```

Comment:
Need to make sure these error conditions are covered by test cases

Reply:
New tests added